substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/api/SharedHostVM.java

```diff
@@ -0,0 +1,155 @@
+
+import com.oracle.graal.pointsto.BigBang;
+import com.oracle.graal.pointsto.PointsToAnalysis;
+import com.oracle.graal.pointsto.meta.AnalysisMethod;
+import com.oracle.graal.pointsto.meta.AnalysisType;
+import com.oracle.graal.pointsto.meta.AnalysisUniverse;
+import com.oracle.graal.pointsto.meta.HostedProviders;
+import jdk.vm.ci.meta.ResolvedJavaType;
+import org.graalvm.compiler.core.common.spi.ForeignCallDescriptor;
+import org.graalvm.compiler.core.common.spi.ForeignCallsProvider;
+import org.graalvm.compiler.java.GraphBuilderPhase;
+import org.graalvm.compiler.nodes.StructuredGraph;
+import org.graalvm.compiler.nodes.graphbuilderconf.GraphBuilderConfiguration;
+import org.graalvm.compiler.nodes.graphbuilderconf.IntrinsicContext;
+import org.graalvm.compiler.options.OptionValues;
+import org.graalvm.compiler.phases.OptimisticOptimizations;
+import org.graalvm.nativeimage.hosted.Feature;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ForkJoinPool;
+import java.util.function.BiConsumer;
+
+public abstract class SharedHostVM implements HostVM {
+    private final List<BiConsumer<Feature.DuringAnalysisAccess, Class<?>>> classReachabilityListeners;
+    private final OptionValues options;
+    protected final ClassLoader classLoader;
+
+    protected final List<BiConsumer<AnalysisMethod, StructuredGraph>> methodAfterParsingHooks = new CopyOnWriteArrayList<>();
+
+    public void addMethodAfterParsingHook(BiConsumer<AnalysisMethod, StructuredGraph> methodAfterParsingHook) {
+        methodAfterParsingHooks.add(methodAfterParsingHook);
+    }
+
+    protected SharedHostVM(OptionValues options, ClassLoader classLoader) {
+        this.options = options;
+        this.classLoader = classLoader;
+        this.classReachabilityListeners = new ArrayList<>();
+    }
+
+    @Override
+    public OptionValues options() {
+        return options;
+    }
+
+    @Override
+    public boolean isRelocatedPointer(Object originalObject) {
+        return false;
+    }
+
+    @Override
+    public void clearInThread() {
+
+    }
+
+    @Override
+    public void installInThread(Object vmConfig) {
+        Thread.currentThread().setContextClassLoader(classLoader);
+    }
+
+    @Override
+    public Object getConfiguration() {
+        return null;
+    }
+
+    @Override
+    public void checkForbidden(AnalysisType type, AnalysisType.UsageKind kind) {
+    }
+
+    @Override
+    public void registerType(AnalysisType newValue) {
+
+    }
+
+    @Override
+    public Optional<AnalysisMethod> handleForeignCall(ForeignCallDescriptor foreignCallDescriptor, ForeignCallsProvider foreignCallsProvider) {
+        return Optional.empty();
+    }
+
+    @Override
+    public GraphBuilderPhase.Instance createGraphBuilderPhase(HostedProviders providers, GraphBuilderConfiguration graphBuilderConfig, OptimisticOptimizations optimisticOpts,
```

Comment:
You can remove this, it is implemented by both subclasses.

Reply:
Fixed